apps/cowswap-frontend/src/modules/orders/containers/CancelledOrderNotification/index.tsx

```diff
@@ -68,22 +50,14 @@ export function CancelledOrderNotification(props: PendingOrderNotificationProps)
         <p>
           Order <strong>{shortenOrderId(orderUid)}</strong>:
         </p>
-        <div>
-          {isSell ? (
-            <>
-              Sell {inputAmountElement} for at least {outputAmountElement}
-            </>
-          ) : (
-            <>
-              Buy {outputAmountElement} for at most {inputAmountElement}
-            </>
-          )}
-        </div>
-        {toAddress && receiver && receiver !== owner && (
-          <div>
-            Receiver: <ExplorerLink id={receiver} label={toAddress} type="address" />
-          </div>
-        )}
+        <OrderSummary
+          kind={order.kind}
+          inputToken={order.inputToken as TokenInfo}
```

Comment:
do we need the casting?

Reply:
Unfortunately yes.
`order.inputToken` has type `Token` from `@uniswap/sdk-core`.
This type for some reason contains `name` and `symbol` as optional fields, but regarding to ERC-20 specification they are required:
https://docs.openzeppelin.com/contracts/3.x/api/token/erc20#ERC20-symbol--

In `TokenInfo` interface I've specified them corresponding to the specification.
Because of it we have to cast the type here.
Eventually, we should get rid of this uniswap type.

Reply:
😢

yes, eventually we should use our `@coprotocol/types`